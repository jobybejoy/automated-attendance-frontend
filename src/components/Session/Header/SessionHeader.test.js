import React from 'react';
import { render, cleanup } from '@testing-library/react';

import SessionHeader from './index';

afterEach(cleanup)

// Unit Test Cases for Session Header Component 
describe('Session Header', () => {

  it('renders to document w/ props', () => {

    const session = {
      course_id: 1,
      course_string: "cs 675",
      session_id: 1,
      session_name: "001",
      course_name: "Software Architecture",
      term: "Fall",
      year: 2021,
      url: "/cs/675/session/001"
    }

    const { getByTestId } = render(<SessionHeader session={session} />);
    // expect(getByTestId('SessionHeader')).toBeInTheDocument();
    expect(getByTestId('SessionHeader')).toMatchSnapshot()
  })

  it('renders to without props', () => {

    // * Expecting the component to render loading state

    // const session = {
    //   course_id: 1,
    //   course_string: "cs 675",
    //   session_id: 1,
    //   session_name: "001",
    //   course_name: "Software Architecture",
    //   term: "Fall",
    //   year: 2021,
    //   url: "/cs/675/session/001"
    // }

    const { getByTestId } = render(<SessionHeader />);
    expect(getByTestId('SessionHeader')).toBeInTheDocument();
  })

})