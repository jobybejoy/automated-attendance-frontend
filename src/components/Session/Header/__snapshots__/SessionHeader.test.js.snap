// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Session Header renders to document w/ props 1`] = `
<div
  class="session_header_container"
  data-testid="SessionHeader"
>
  <div
    class="session_meta_details"
  >
    <span>
      <b>
        CS 675
      </b>
       
    </span>
    <span>
      001
    </span>
    <span>
      Fall 2021
    </span>
  </div>
  <h1
    class="course_name"
  >
    Software Architecture
  </h1>
</div>
`;
